### PR TITLE
[Silabs] Enabling Sleepy on 917 SoC device

### PR DESCRIPTION
**Pull request by chirag-silabs**
M4 sleep was not enabled on the 917 SoC

1. Enabling the 917 SoC sleep whenever the device go to the idle task.
2. Reinit silabs logs and lcd after every wakeup on the MCU
3. Adding Alarm based periodic wakeup as an extra argument which can be enabled/disabled from the user.
4. Update matter_support

### CODE REVIEW

File: third_party/silabs/SiWx917_sdk.gni [hunk 1]
```diff
@@ -24,6 +24,14 @@ import("silabs_board.gni")
 
 examples_plat_dir = "${chip_root}/examples/platform/silabs/SiWx917"
 
+declare_args() {
+  # Enable the Alarm Based Wakeup for 917 SoC when sleep is enabled
+  alarm_based_wakeup = false
```
Comment (jmartinez-silabs) [inline third_party/silabs/SiWx917_sdk.gni:29]: maybe prefixing with `si91x_` would clarify what they are used for

File: examples/platform/silabs/SiWx917/SiWx917/sl_wlan_config.h [hunk 2]
```diff
@@ -28,7 +28,16 @@
 //! Disable feature
 #define RSI_DISABLE 0
 
-#define SI91X_LISTEN_INTERVAL 0
+#define IVT_OFFSET_ADDR 0x8202000         /*<!Application IVT location VTOR offset>        */
```
Comment (jmartinez-silabs) [inline examples/platform/silabs/SiWx917/SiWx917/sl_wlan_config.h:31]: is this file the right place for those defines? It isn't related to wlan configs

File: examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c [hunk 3]
```diff
@@ -198,6 +208,42 @@ sl_status_t join_callback_handler(sl_wifi_event_t event, char * result, uint32_t
...
+        if (RSI_NPSSGPIO_GetPin(SL_BUTTON_BTN0_PIN))
+        {
+#ifdef DISPLAY_ENABLED
+            // if LCD is enabled, power down the lcd before setting the M4 to sleep
+            sl_si91x_hardware_setup();
+#endif
+            btn0_pressed = false;
+            /* Configure RAM Usage and Retention Size */
+            sl_si91x_m4_sleep_wakeup();
+            silabsInitLog();
```
Comment (jmartinez-silabs) [inline examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c:241]: why are we re-initing the logs here?

File: examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c [hunk 4]
```diff
@@ -198,6 +208,42 @@ sl_status_t join_callback_handler(sl_wifi_event_t event, char * result, uint32_t
...
+/******************************************************************
+ * @fn   M4_sleep_wakeup()
+ * @brief
+ *       M4 going to sleep
+ *
+ * @param[in] None
+ * @return
+ *        None
+ *********************************************************************/
+void M4_sleep_wakeup()
```
Comment (jmartinez-silabs) [inline examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c:222]: I am not a fan of the function name. It doesn't fit the rest of the apis

File: third_party/silabs/SiWx917_sdk.gni (hunk 1 above)
Comment (chirag-silabs) [reply third_party/silabs/SiWx917_sdk.gni]: added `si91x_` prefix for these alarm based build options

File: examples/platform/silabs/SiWx917/SiWx917/sl_wlan_config.h (hunk 2 above)
Comment (chirag-silabs) [reply examples/platform/silabs/SiWx917/SiWx917/sl_wlan_config.h]: Yeah, removed these defines. Already these defines are added in the matter support

File: examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c (hunk 3 above)
Comment (chirag-silabs) [reply examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c:242]: In 917 SoC once M4 goes to sleep it, it disables everything so we have to re init the ones which we need like btn, logs and all

File: examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c (hunk 4 above)
Comment (chirag-silabs) [reply examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c]: renamed the API to `sl_wfx_host_si91x_sleep_wakeup()`

**Review by jmartinez-silabs — APPROVED** [review]

**Review by saurabhst — APPROVED** [review]

**Pull request merged**